bufrsnd/rrfs_bufr.fd/PROF_FV3SAR_NET.f

```diff
@@ -764,27 +764,28 @@ SUBROUTINE PROF_FV3SAR_NET(filedyn,filephys,datestr,ITAG,INCR,
         LRECPR=4*(8+9+LCL1ML*LM+LCL1SL)
 ! former parameter statements
 
-        if (allocated(FPACK)) deallocate(FPACK); allocate(FPACK(NWORDM))
-        if (allocated(PRODAT)) deallocate(PRODAT);
-     &                  allocate(PRODAT(NWORDM))
-        if (ALLOCATED(DUM)) deallocate(DUM);
-     &                          allocate(DUM(IM,JM,4))
-        if (ALLOCATED(DUMMY)) deallocate(DUMMY);
-     &                          allocate(DUMMY(IM,JM))
-        if (ALLOCATED(DUMMY2)) deallocate(DUMMY2);
-     &                          allocate(DUMMY2(IM,JM))
-
-        if (ALLOCATED(DUM3D)) deallocate(DUM3D);
-     &                          allocate(DUM3D(NUMSTA,LM))
-        if (ALLOCATED(DUM3D2)) deallocate(DUM3D2);
-     &                          allocate(DUM3D2(NUMSTA,LM))
-        if (ALLOCATED(DUM3D3)) deallocate(DUM3D3);
-     &                          allocate(DUM3D3(NUMSTA,LM))
-        if (ALLOCATED(DUM3D4)) deallocate(DUM3D4);
-     &                          allocate(DUM3D4(NUMSTA,LM))
-
-        if (ALLOCATED(LMH)) deallocate(LMH);
-     &                          allocate(LMH(IM,JM))
+        if (allocated(FPACK)) deallocate(FPACK)
+        allocate(FPACK(NWORDM))
```

Comment:
Just curious, if FPACK is already allocated, why do we want to deallocate and allocate it again?

Reply:
@mark-a-potts - Your current solution is safer because it exactly reproduces the existing behavior.  If you change it to what you proposed, the logic is different and may produce incorrect behavior.  Of course, the existing behavior may also be incorrect.  But there's no way to know that.  Unless someone can produce a robust way to test the alternative, I think your current implementation is best.

Reply:
Thank @christopherwharrop-noaa and @mark-a-potts for the discussions here. I agree that since Mark does not actually change this source code, we can leave them as-is.